### ISSUE TITLE
Get packages from the Umbraco marketplace 

### ISSUE THREAD

**Issue by prjseal** [prjseal/Package-Script-Writer#27]
Currently we pull the packages feed from Our Umbraco but that is out of date now and needs to come from the Umbraco Marketplace.

We can get the data from the Umbraco Marketplace in batches of 50 using [this API request](https://api.marketplace.umbraco.com/api/v1.0/packages?orderBy=MostDownloads&fields=numberOfNuGetDownloads,authors,packageType,licenseTypes,iconUrl,minimumUmbracoVersionNumber,maximumUmbracoVersionNumber,isCertifiedToWorkOnUmbracoCloud,isPromoted,isPartner,isHQ,isHQSupported,id,title,description,packageId,tags,umbracoMajorVersionsSupported,iconDominantColor&pageSize=50&pageNumber=1).

This is what a result looks like. We need a c# model to match it.

```json
{
    "authors": "Jumoo",
    "description": "Push and Pull, Content, Media and Settings between Umbraco installations from the comfort of the Backoffice UI",
    "iconDominantColor": "#101C2F",
    "iconUrl": "icons/usync.complete.png",
    "id": "27028c6c-e646-4e81-ac8c-38b8343545d7",
    "isHQ": false,
    "isHQSupported": false,
    "isPromoted": false,
    "isPartner": false,
    "latestVersionNumber": "12.1.0",
    "licenseTypes": [
        "Purchase",
        "Subscription"
    ],
    "minimumUmbracoVersionNumber": "9.0.0",
    "numberOfNuGetDownloads": 530636,
    "packageId": "uSync.Complete",
    "packageType": "Package",
    "tags": [
        "deployment"
    ],
    "title": "uSync.Complete",
    "umbracoMajorVersionsSupported": [
        9,
        10,
        11,
        12
    ]
},
```

The URL for the package will be a combination of https://marketplace.umbraco.com/package/ and the **packageId** field

The URL for the icon is a combination of https://marketplace.umbraco.com/ and the **iconUrl** field

The number of downloads comes from the numberOfNuGetDownloads field

**Comment by hannakfoo** [comment]
Hi Paul, you mean something like this? https://gist.github.com/hannakfoo/6e6e76175153e921b0b48e67b183bb85

**Comment by prjseal** [comment]
Hi yes, kind of but we only care about the full url of the icon and package when we are doing the rendering part.

What we need first is to query all of the packages into a list of objects similar to this and then cache it and render it out like the existing ones are.

It's best to see how we have done it already using Our Umbraco and then try to create a different service for the Marketplace.

**Comment by prjseal** [comment]
This has been resolved now thanks to @hannakfoo in this PR #28
